README.md

```diff
@@ -17,95 +17,177 @@ julia> ] add https://github.com/andrewrosemberg/PortfolioOpt.jl.git
+ - Conditional expected return (`ConditionalExpectedReturn`) - also called Conditional Value at Risk (CVAR) or (Expected Shortfall);
+ - Expected utility (`ExpectedUtility`) which computes the expected value of a specified (hopefully concave) utility function (`ConcaveUtilityFunction`):
+    - the only implemented one is the piece-wise concave utility function `PieceWiseUtility`.
+
+Given that `AmbiguitySet`s might be sets of distributions, it is necessary to determine which distribution to use in the definition of the `PortfolioRiskMeasure`. This choice can be imposed by the user through his level of robustness (type `Robustness`):
+ - `EstimatedCase` if dealing with `CenteredAmbiguitySet`s and the user doesn't want to add any robustness (default);
+ - `WorstCase` if the decision maker wants to use the worst case distribution in the ambiguity set.
+
+The `PortfolioRiskMeasure`s can be used to define both the `RiskConstraint`s and the `ObjectiveTerm`s in a `PortfolioFormulation` that can be parsed into details of a `JuMP.Model` using the `portfolio_model!` function.
+
+In addition, `ObjectiveTerm`s can also be `ConeRegularizer`s defined by a cone set (e.g. `norm-2`) and a linear transformation (default Identity).
+
+## VolumeMarket
+The `portfolio_model!` modifies an existing model `JuMP.Model` with decision variables already created and which, ideally, are already bounded by budget and bound constraints. In order to help users define market constraints, fees and clearing processess, his package also implements an interface with `OptimalBids.jl` (a framework for working with generic markets) through a simple market type called `VolumeMarket`.
+
+`VolumeMarket` represents market models that only allow the strategic agent to bid at market price, thus their decision is restricted to the amount/volume traded of each available assets.
+
+The current implementation allows the user to specify:
+ - `budget::Real`: Total amount of resources/volume that can be invested (usually the sum the vector of individual invested amounts or the 1-norm of it);
+ - `volume_fee::Real`: Cost per unit of volume invested;
+ - `allow_short_selling::Bool`: If true allows decision variables to be negative; 
+ - `risk_free_rate::Real`: Risk free return (known return of the money not invested).
+
+Once an instance of a `VolumeMarket` is defined, one can call `market_model` to create a `JuMP.Model` with the equivalent constraints, objective terms and variables created. Moreover, after the strategic objective terms and constraints are added on top of this model, it can be passed to the `change_bids!` together with the `VolumeMarket` object to modify the `volume_bids::Vector{Real}`. Alternetivelly, `change_bids!` can receive the already calculated bids (if chosen elsewhere) or even just the `PortfolioFormulation`, leaving the work of creating the `JuMP.Model` and adding all constraints and objective terms (market based or strategy based) to this function.
```

Comment:
```suggestion
Once an instance of a `VolumeMarket` is defined, one can call `market_model` to create a `JuMP.Model` with the equivalent constraints, objective terms and variables created. Moreover, after the strategic objective terms and constraints are added on top of this model, it can be passed to the `change_bids!` together with the `VolumeMarket` object to modify the `volume_bids::Vector{Real}`. Alternatively, `change_bids!` can receive the already calculated bids (if chosen elsewhere) or even just the `PortfolioFormulation`, leaving the work of creating the `JuMP.Model` and adding all constraints and objective terms (market based or strategy based) to this function.
```